### PR TITLE
Vpc flow without lambda function

### LINKED ISSUE #6: VPCFlow without lambda function
update cloudformation template to collect and write vpcflow logs to an s3 bucket without the usage of a lambda function

### PR DESCRIPTION
**Pull request by sonofagl1tch**
Closes #6

modified template to write vpc flow logs directly to s3 rather than through a lambda function

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mgmacias95 — APPROVED** [review]
Hello @sonofagl1tch,

VPC Flow logs are written to the S3 bucket correctly but I'm not getting alerts in Wazuh. I think the reason could be that the path specified in the configuration is different than the path created in the S3 bucket:

imagen
```xml
  <bucket type="custom">
   <name>detonationlab-s3bucketvpcflow-2ok03g4b33e4</name>
<path>flowlogs</path>
   <access_key>xxxxxx</access_key>
   <secret_key>xxxxx</secret_key>
  </bucket>
```
In my configuration I have `flowlogs`, in my bucket I have `vpcflowlogs`. 

I'm approving this PR, since VPC Flow logs are written in S3 buckets (this is what this PR does), but I'd like this to be reviewed and fixed in a separate PR.

Best regards,
Marta

**Comment by mgmacias95** [comment]
Hi @sonofagl1tch,

I found out why Wazuh is not able to read logs using this. I have opened an issue in the wazuh repository (https://github.com/wazuh/wazuh/issues/1516). **This PR should NOT be merged until the Wazuh team fixes this issue**.

Best regards,
Marta

**Comment by mgmacias95** [comment]
Hello @sonofagl1tch,

I was trying to test this again and I wasn't able to deploy it. I get the following error:
imagen
It seems it's necessary to remove the lambda function previously used to retrieve VPC logs.

Best regards,
Marta

**Comment by sonofagl1tch** [comment]
> Hello @sonofagl1tch,
> 
> I was trying to test this again and I wasn't able to deploy it. I get the following error:
> imagen
> It seems it's necessary to remove the lambda function previously used to retrieve VPC logs.
> 
> Best regards,
> Marta

found that the lambda resources were still in the tempate. once i removed them the template loaded successfully.

File: awsDetonationLab.template [hunk 1]
```diff
@@ -876,107 +876,27 @@
       }
     },
     "FlowLogUpload": {
-      "Type": "AWS::Lambda::Function",
+      "Type": "AWS::EC2::FlowLog",
       "Properties": {
-        "Handler": "index.handler",
-        "MemorySize": 256,
-        "Role": {
+        "DeliverLogsPermissionArn": {
```
Comment (mgmacias95) [inline awsDetonationLab.template]: According to the [AWS::EC2::FlowLog documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-flowlog.html#w2ab1c21c10d102c40b9):
> If you specify `LogDestinationType` as `s3`, do not specify `DeliverLogsPermissionArn` or `LogGroupName`. 

Which means this field is no longer necessary and must be removed.

**Review by mgmacias95 — CHANGES_REQUESTED** [review]

**Comment by sonofagl1tch** [comment]
@mgmacias95 this is ready for review again

File: awsDetonationLab.template (hunk 1 above)
Comment (sonofagl1tch) [reply awsDetonationLab.template]: removed vpc lambda related resources from template

Comment (mgmacias95) [reply awsDetonationLab.template]: Aren't those vpc lambda related too?
https://github.com/sonofagl1tch/AWSDetonationLab/blob/a681a7465d78f038f6aba1ee54a4cdcbfe185d10/awsDetonationLab.template#L847-L851
https://github.com/sonofagl1tch/AWSDetonationLab/blob/a681a7465d78f038f6aba1ee54a4cdcbfe185d10/awsDetonationLab.template#L879-L917
https://github.com/sonofagl1tch/AWSDetonationLab/blob/a681a7465d78f038f6aba1ee54a4cdcbfe185d10/awsDetonationLab.template#L867-L878

**Review by mgmacias95 — APPROVED** [review]
Hello @sonofagl1tch,

I think the code is ready in our side, but further testing needs to be done when Wazuh releases the new AWS integration including support for this PR.

Best regards,
Marta

**Comment by mgmacias95** [comment]
Hi there,

I have tested this PR using Wazuh v3.8.0. The following additional permissions need to be added to the user the detonation lab creates:
```javascript
       {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "ec2:DescribeFlowLogs",
            "Resource": "*"
        },
```

Best regards,
Marta

**Comment by mgmacias95** [comment]
A new policy with the necessary permissions has been added in commit https://github.com/sonofagl1tch/AWSDetonationLab/pull/21/commits/cc2149a9b7e8854428af0ec686958ad7ff5351a0. 

The PR works as expected except for issue https://github.com/sonofagl1tch/AWSDetonationLab/issues/61 which I think must be fixed in a separate PR.

@sonofagl1tch this is ready to merge! 😄

**Comment by sonofagl1tch** [comment]
The new Iam policy you created is over privileged.  Using an asterisk for resources means that any user/role/group that uses that policy can run those commands against any reside in your account. I suggest looking at how I created previous access policies in the template so we can restrict access to only detonation lab related resources. Let's talk offline about this before the commit.

**Comment by mgmacias95** [comment]
Hello @sonofagl1tch,

I tried modifying the policy to make it resource specific as you suggested and got the following warning:
imagen
Which roughly translates as _"the action needs access to all resources_".

Best regards,
Marta

**Comment by sonofagl1tch** [comment]
after researching through VPC documentation, the new feature to all VPC flow to write directly to S3 does not support an option to restrict the ec2:DescribeFlowLogs api call to specific flows. a feature request to the vpc flow team will need to be opened to correct this over priveleged feature. This PR should be good to approve and merge.

**Pull request merged**